easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -202,7 +202,7 @@ def get_toolchain_hierarchy(parent_toolchain):
     toolchain_hierarchy = [parent_toolchain]
     # use a queue to handle a breadth-first-search of the hierarchy,
     # which is required to take into account the potential for multiple subtoolchains
-    bfs_queue = [parent_toolchain]
+    bfs_queue = [dict(parent_toolchain)]
```

Comment:
Do it this way so we don't interfere with the parent toolchain when adding capabilities